File: hc-mdm-web/src/main/webapp/resources/app/view/bassupplycategory/BasSupplyCategoryController.js
/**
 * Description: Controller
 * All rights Reserved, Designed By Hc
 * Copyright:   Copyright(C) 2014-2015
 * Company:     Wonhigh.
 * @author:     Administrator
 * @date:  2015-04-15 18:24:21
 * @version 1.0.0
 */
Ext.define('Hc_mdm.view.bassupplycategory.BasSupplyCategoryController', {
    extend: 'Hc_Common.view.BaseSimplePageController',
    alias: 'controller.bassupplycategory',
    
    // 网格列编辑后
	onGridAfterEdit : function(editor, e) {
	    var me = this;
	    store = me.lookupReference('mastergrid').store;
	    if (e.field == 'categoryName' && e.value != ''
	            && e.value != null) {
	        for (var i = 0; i < store.totalCount; i++) {
	            if (e.rowIdx == i)
	                continue;
	            var model = store.getAt(i);
	            if (model.get('categoryName') == e.value) {
	                Hc.alert("类别名称不能重复");
	                e.record.set('categoryName', '');
	                return;
	            }
	        }
	    }
	}
  });

File: hc-mdm-web/src/main/java/com/hc/scm/mdm/web/controller/BasSupplyCategoryController.java
package com.hc.scm.mdm.web.controller;

import com.hc.scm.common.base.service.BaseCrudService;
import com.hc.scm.common.base.web.BaseCrudController;
import com.hc.scm.mdm.dao.entity.BasSupplyCategory;
import com.hc.scm.mdm.service.BasSupplyCategoryService;
import javax.annotation.Resource;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

/**
 * Description: 请写出类的用途
 * All rights Reserved, Designed By hc* Copyright:   Copyright(C) 2014-2015
 * Company:     Wonhigh.
 * @author:     Administrator
 * @date:  2015-04-15 16:18:21
 * @version 1.0.0
 */
@Controller
@RequestMapping("/bas_supply_category")
public class BasSupplyCategoryController extends BaseCrudController<BasSupplyCategory> {
    @Resource
    private BasSupplyCategoryService basSupplyCategoryService;

    @Override
    public BaseCrudService init() {
        return basSupplyCategoryService;
    }
}